src/Rules/IRuleInfo.cs

```diff
@@ -120,13 +120,19 @@ public interface IRuleInfo
         /// </summary>
         string Description { get; }
 
+        /// <summary>
+        /// List of tags. Can be empty. Will not be null.
+        /// </summary>
```

Comment:
I've changed `RuleInfo` so `Tags`, `DescriptionSections` and `EducationPrinciples` will return empty collections rather than nulls. That will make them easier to work with in the rest of the client code.